contributing/guide-api-promotion.md

```diff
@@ -0,0 +1,97 @@
+# Promoting an API
+
+This guide describes the process for promoting an API in Crossplane from one
+version to the next. This process should be handled with care over multiple
+releases, as it has proven fairly easy for Crossplane's API promotions of the
+past to be problematic for both upgrades and downgrades, as demonstrated by
+[#6148], [#5932], and [#4400].
+
+An explanation of Crossplane's feature lifecycle can be found in the [Feature
+Lifecycle] docs page.
+
+## Core Issue
+
+The root cause we have encountered in past problematic API promotions is that we
+try to drop a particular version from a CRD while there may still be resources
+of that version stored in etcd. Kubernetes does not allow this action because it
+can result in possible data loss.
+
+Note that this can happen on both upgrades as well as downgrades. Let's briefly
+summarize one common downgrade path where we've encountered this issue:
+
+1. A Crossplane API is promoted to Beta and the storage version for the CRD is
+   updated to Beta in the same release
+1. User installs this new Crossplane version and creates a Beta resource that is
+   stored in etcd as Beta
+1. User downgrades to the previous Crossplane version where the CRD's Beta
+   version did not exist
+1. The Crossplane init container tries to update the CRD back to the Alpha
+   version, which drops the Beta version
+1. The Kubernetes API server rejects this action because there are still Beta
+   resources stored in etcd according to the CRD's [`status.storedVersions`]
+   field
```

Comment:
```suggestion
1. The Kubernetes API server rejects this action because there might be still Beta
   resources stored in etcd according to the CRD's [`status.storedVersions`]
   field
```

Reply:
"it doesn't know better"